### PR TITLE
Add TS path aliases, allow HMR and environment variables validation using zod

### PR DESCRIPTION
**Pull request by saenyakorn**
# What I did
 
- Allow us to use Typescript path alias (Already test on dev, build and start command)
- Enable HMR for development
- Validate environment variables using Zod before starting the application

### CODE REVIEW

**Comment by saenyakorn** [comment]
Here's the example of env validation 

Screenshot 2566-05-17 at 12 12 03

**Review by SaltyAom — APPROVED** [review]
Good stuff, but I would have an urge for path aliasing, might need discussion.

Leaving this to other reviewer.

**Comment by saenyakorn** [comment]
Yeah, it's weird that it need `tsc-alias`. But I only have much time to fix the issue. So, I decide to use it first.

File: tsconfig.json [hunk 1]
```diff
@@ -1,18 +1,27 @@
 {
   "compilerOptions": {
-    "target": "ESNext",
-    "module": "ESNext",
-    "lib": ["ESNext"],
+    "target": "esnext",
+    "module": "commonjs",
```
Comment (Leomotors) [inline tsconfig.json:4]: Why commonjs?

Comment (SaltyAom) [reply tsconfig.json:4]: Because Node ESModule compatibility is bad

**Comment by dtinth** [comment]
thank you for the suggestion, zod env validation has been added here and i added you as a co author 78189c3b3caf1e6981c024105763a11ad91162ea

**Comment by saenyakorn** [comment]
Close this since it outdated

**Pull request closed without merging**